### ISSUE TITLE
Append Codelist

### ISSUE THREAD

**Issue by romefi** [opengisch/projectgenerator#15]
Possiblity to append a codelist from a model if available. Think ili2pg allows that.

**Comment by m-kuhn** [comment]
Can you share an example model for this?

**Comment by romefi** [comment]
Sure.

Model: http://models.geo.admin.ch/BAFU/KbS_V1_3.ili
Codelist: http://models.geo.admin.ch/BAFU/KbS_Codetexte_V1_3.xml

**Comment by pramisters** [comment]
I agree with Romedi that the import of XML-catalogues is a very important feature that should be covered by the projectgenerator.
The example that Romedi provided is a bad one because the codelists are not modelled as proposed by swisstopo using the catalogue objects (https://models.geo.admin.ch/CH/CHBase_Part3_CATALOGUEOBJECTS_20110830.ili) of the "Basismodule". 

A better example is: 
Model: https://models.geo.admin.ch/BLW/Nutzungsflaechen_V1_4.ili
Catalogue: https://models.geo.admin.ch/BLW/Nutzungsflaechen_V1_4_Kataloge_20170322.xml

To ensure that a later update of the XML-catalogues works without destroying existing data (foreign keys e.g.) a few options of ili2pg are mandatory:
--coalesceCatalogueRef
--createBasketCol
--importTid

**Comment by m-kuhn** [comment]
* How do we deal with baskets?
  * One basket is created per dataset (a dataset is a catalogue or payload data)
  * For new data, one needs to specify the basket id which is hard to know
* Update of catalogue data??

**Comment by m-kuhn** [comment]
@romefi 
@pramisters 

now that we have the `--createBasketCol`, do we need something else for catalogue support?
Does import of the catalog need to be integrated in the user interface (and if yes, how)?

*Note: Handling of default values for the basket col are to be discussed separately, let's focus on the catalogue in here.*

**Comment by romefi** [comment]
As my understanding ist, if the Catalogue is well-definied, it can be imported with the "Import Interlis File" and a new basket should be generated. The format of Catalogues is generally .xml. So if .xml-Files are allowed in the Import interface, problem might be solved.

@pramisters knows ili2pg better than me, maybe he has some adjustments

**Comment by m-kuhn** [comment]
What about this:

* on the generate and on the import interface, there is an "additional files" section that allows importing one or more xml files (catalogs)
* with the generate option, a "standard" basket gets created and set as default basket
* with the import option, a `--dataset` parameter is specified and set as default basket
* update of catalogs has to be done manually (for now)
* switching between baskets with a user interface can be handled separately https://github.com/opengisch/projectgenerator/issues/103

**Comment by zigertiger** [comment]
to me, this sounds like an elegant solution for the Moment (since catalogue data is more tightly tied to a model than "ordinary" transfer data).

nevertheless, the values for --dataset must be specified by the user, not generic

**Comment by m-kuhn** [comment]
Add .xml to allowed import formats for a first step

**Comment by romefi** [comment]
> Add .xml to allowed import formats for a first step

May add .itf as well to import data in Interlis 1
